packages/venia-ui/lib/targets/BabelRouteInjectionPlugin.js

```diff
@@ -0,0 +1,111 @@
+                    state.routes = [];
+                    const seenPatterns = new Map();
+                    const seenModules = new Map();
+                    const requests = this.opts.requestsByFile[this.filename];
+                    for (const request of requests) {
+                        const { requestor } = request;
+                        for (const route of request.options.routes) {
+                            const routeWithRequestor = { ...route, requestor };
+                            const seenPattern = seenPatterns.get(route.pattern);
+                            if (!seenPattern)
+                                seenPatterns.set(
+                                    route.pattern,
+                                    routeWithRequestor
+                                );
+
+                            const seenExactPattern =
+                                seenPattern &&
+                                seenPattern.exact === route.exact;
+
+                            const seenModule = seenModules.get(route.path);
+                            if (!seenModule)
+                                seenModules.set(route.path, routeWithRequestor);
+
+                            if (!seenModule && !seenExactPattern) {
+                                // a nice, fresh route!
+                                state.routes.push(route);
+                            } else if (seenExactPattern) {
+                                throw new Error(
+                                    `@magento/venia-ui: Conflict in "routes" target. "${
+                                        request.requestor
+                                    }" is trying to add a route ${JSON.stringify(
+                                        route
+                                    )}, but "${
+                                        seenExactPattern.requestor
+                                    }" has already declared that route pattern: ${JSON.stringify(
+                                        seenExactPattern
+                                    )}`
+                                );
+                            } else if (seenModule) {
+                                throw new Error(
+                                    `@magento/venia-ui: Conflict in "routes" target. "${
+                                        request.requestor
+                                    }" is trying to add a route ${JSON.stringify(
+                                        route
+                                    )}, but "${
+                                        seenModule.requestor
+                                    }" has already declared another pattern for that same module: ${JSON.stringify(
+                                        seenModule
+                                    )}`
+                                );
+                            }
+                            // if we got here, both have been seen, so this is an exact duplicate;
```

Comment:
Would we reach this statement by bad configuration? If so might be nice to give the developer a little message explaining there is a duplicate

Reply:
We could do that, but I don't know what circumstances would cause it. I do know that if two different interceptors add the exact same route, the intent is clearly to add the route once. I don't think that removing a duplicate is surprising behavior. And since I don't know how, or if, duplicate requests will come in, this seemed like a safe choice.

I want to report all ambiguities to the developer, but within reason or we get too chatty. This one seemed like it's not ambiguous what the desired result is.

Reply:
When we're discussing routes here, we're discussing paths to a module ("Babel routes"), correct?

Not URL patterns?